substrate-sgx/externalities/src/codec_impl.rs

```diff
@@ -2,14 +2,15 @@
 //!
 //! This is little workaround, as `Encode` and `Decode` can't directly be implemented on `HashMap`.
 
-use crate::{SgxExternalitiesDiffType, SgxExternalitiesType};
 use codec::{Decode, Encode, Input};
-use sgx_serialize::{DeSerializable, DeSerializeHelper, Serializable, SerializeHelper};
+use serde::{de::DeserializeOwned, Serialize};
 use std::{vec, vec::Vec};
 
+use crate::{SgxExternalitiesDiffType, SgxExternalitiesType};
+
 impl Encode for SgxExternalitiesType {
 	fn encode(&self) -> Vec<u8> {
-		encode_with_serialize(self)
+		serde_json::to_vec(&self).unwrap()
```

Comment:
Use `serde_json` for now, as discussed in #26

Reply:
I can agree with serde, but is the unwrap necessary?

Reply:
the unwrap is not ideal, but what would be the alternative? the signature of the function is given, as it is an implementation of `Encode`.

Reply:
I'm also unsure. Here https://github.com/integritee-network/sgx-runtime/blob/feature/fm-externalities-deterministic-serialization/substrate-sgx/externalities/src/codec_impl.rs#L36-L41 we're returning an empty vec and an error log. I think that should be better than a panic. What do you think?

Reply:
I thought about that option too. In my opinion it is worse to swallow the error by just returning an empty vector. If this serialization fails, there is something programmatically wrong, and I think we should explicitly abort in that case. Otherwise we have some edge-cases in the client where the encoded state is an empty vector, which might still be valid, depending on how we treat it?

Reply:
ahm but anyway, you're right. The real problem is that we have two different implementations, which should not be the case!

Reply:
Agreed on your take about the unwrap: If we can't encode a state. something is seriously wrong. I'm fine with the unwrap (s).

Reply:
So we had the empty vector behavior before.. so I guess I'll just leave it like that, even though I think it's potentially problematic (as could the `unwrap()` also be 🤔 )

Reply:
Acutally I'm against leaving it like that. Because returning an empty state will mess up a lot of things and should not be accepted. So either we need to check for it in the enclave (which can be easily forgotten) and react upon that or we panic here, which results in the need of a restart. 

So I think we should create an issue to be aware of these unwraps, and unless we decide upon a better empty state handling, we leave it at that.

Reply:
I've removed the `unwrap()` to preserve the behavior. I think cases can be made for both options, but this way we at least don't change any existing caller code that might rely on this behavior.